crates/uv/src/commands/project/lock.rs

```diff
@@ -122,7 +122,28 @@ pub(super) async fn do_lock(
         .map(UnresolvedRequirementSpecification::from)
         .collect();
     let constraints = vec![];
-    let overrides = vec![];
+
+    let mut overrides: Vec<UnresolvedRequirementSpecification> = vec![];
+    workspace.packages().values().for_each(|member| {
+        if let Some(override_dependencies) = member
+            .pyproject_toml()
+            .tool
+            .as_ref()
+            .and_then(|tool| tool.uv.as_ref())
+            .and_then(|uv| uv.override_dependencies.as_ref())
+        {
+            for override_dependency in override_dependencies {
+                let mut req = pypi_types::Requirement::from(override_dependency.clone());
+                // retain override origin for conflict reporting
+                req.origin = Some(pep508_rs::RequirementOrigin::Project(
+                    member.root().clone(),
+                    member.project().name.clone(),
+                ));
+                overrides.push(UnresolvedRequirementSpecification::from(req));
+            }
+        }
+    });
+    let overrides = overrides;
```

Comment:
I think this can be removed?

Reply:
> I think this can be removed?

Wanted to make it explicit that it's now immutable

Reply:
We don't usually do this in our code. It would add a lot of noise.

Reply:
fyi i removed and also changed the iteration style to align more with what we do elsewhere